cmd/kubectl-gadget/top/top.go

```diff
@@ -17,31 +17,41 @@ package top
 import (
 	"fmt"
 	"strings"
-	"sync"
 
 	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
 
 	"github.com/spf13/cobra"
 )
 
-var (
-	params utils.CommonFlags
-	mutex  sync.Mutex
+type CommonTopFlags struct {
+	OutputInterval int
+	MaxRows        int
+	SortBy         string
+}
 
-	outputInterval int
-	maxRows        int
-	sortBy         string
-)
+func NewTopCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "top",
+		Short: "Gather, sort and periodically report events according to a given criteria",
+	}
 
-var TopCmd = &cobra.Command{
-	Use:   "top",
-	Short: "Gather, sort and periodically report events according to a given criteria",
-}
+	cmd.AddCommand(newBlockIOCmd())
+	cmd.AddCommand(newFileCmd())
+	cmd.AddCommand(newTCPCmd())
+	cmd.AddCommand(newEbpfCmd())
 
-func addTopCommand(command *cobra.Command, defaultMaxRows int, sortBySlice []string) {
-	command.Flags().IntVarP(&maxRows, "max-rows", "r", defaultMaxRows, "Maximum rows to print")
-	command.Flags().StringVarP(&sortBy, "sort", "", sortBySlice[0], fmt.Sprintf("Sort column, possible values are: %s", strings.Join(sortBySlice, ", ")))
+	return cmd
+}
 
-	utils.AddCommonFlags(command, &params)
-	TopCmd.AddCommand(command)
+func addCommonTopFlags(
+	command *cobra.Command,
+	commonTopFlags *CommonTopFlags,
+	commonFlags *utils.CommonFlags,
+	defaultMaxRows int,
+	sortBySlice []string,
+) {
+	command.Flags().IntVarP(&commonTopFlags.MaxRows, "max-rows", "r", defaultMaxRows, "Maximum rows to print")
+	command.Flags().StringVarP(&commonTopFlags.SortBy, "sort", "", sortBySlice[0], fmt.Sprintf("Sort column, possible values are: %s", strings.Join(sortBySlice, ", ")))
```

Comment:
I know it wasn't introduced in this PR, but `possible values are` seems too verbose. What about?

```suggestion
	command.Flags().StringVarP(&commonTopFlags.SortBy, "sort", "", sortBySlice[0], fmt.Sprintf("Sort column (%s)",  strings.Join(sortBySlice, ", ")))
```